files/demo.hledger

```diff
@@ -0,0 +1,178 @@
+# It is a plain text file, so it is easy to edit and manipulate, and you can use
+# search and replace functionality in your text editor
+
+# Everything after '#', ';' or '*' are comments, so they are not read by hledger,
+# except in some cases to be discussed later
+
+
+# To start configuring our journal, it is a good idea to define a default currency
+D USD 1,000.00  ; Format - D Currency amountFormat
+
+
+# Now let's declare our main accounts. The order they appear here is the other they will
+# show on the reports. We need to tell hledger what is the name of each account using the
+# following code: A=Asset, L=Liabilities, R=Revenues, X=Expenses, E=Equity.
+# We will use the default name, but you can change them if you like
+
+# Attention! Here and in some other places you may need at least 2 spaces, instead of 1
+
+account Asset                  A  ; Format - account name  Code  <- 2 spaces at least between name and code
+account Liabilities            L
+account Revenue                R
+account Expenses               X
+account Equity                 E
+
+
+# Let's declare some other accounts, so they will appear in the desired order
+# You can organize them in a tree structure using ':' to indicate a a parent-child relation account
+
+account Asset:Cash
+account Asset:Investments
+account Asset:Real Estate
+account Liabilities:Credit Card
+account Liabilities:Mortgage
+
+
+# Possibly you will not start from scratch. You will want to import your data from a previous software that you
+# want to adapt for the powerful hierarchical account system hledger provides. For that we will use alias so that
+# whenever hledger sees an old account, it will substitute for its new name
+
+alias jpm=Asset:Bank:JP Morgan Chase                     ; Format - alias OLD=NEW
+alias visa=Liabilities:Credit Card:Visa Platinum         ; Substitute 'visa' for 'Liabilities:Credit Card:Visa Platinum'
+
+
+# Now we will finally start adding transactions to our ledger. For that, you will need to understand the format. Don't
+# worry if it seems complicated. You will quickly be used to it, and hledger offers some great tools to make that
+# process easier with an interactive filling app with auto-complete and tools to automatically import data from your
+# bank or any other source. It may even automatically fill the data for you if it finds a similar older transaction!
+# If You you are curious, google hledger-iadd or hledger-autosync for some examples
+
+# A transaction is a financial event that has multiple posting showing where the money came and where it goes to.
+# The main property of a transaction is that it always need to balance, or in other words, it values should sum 0.
+# This property is called 'double-entry accounting'. It may seem hard to use, but hledger uses some ingenious approach
+# that makes it almost as simple as a regular 'single-entry-accounting' that you may be used to do.
```

Comment:
(why ? maybe: "because every transaction must have two or more balanced postings."  ?)

Reply:
In fact I didn't mentioned why. I was thinking about the implicit amount feature. If you use it with only 2 postings, it appears to be single entry accounting because one is the Cash/Bank account an the other looks like a 'Category' and not another account. What do you think I should do about this paragraph?

Reply:
Your call, it's just a thought